### ISSUE TITLE
Settings don't persist in Brave browser

### ISSUE THREAD

**Issue by melink14** [melink14/rikaikun#333]
Probably Brave doesn't support storage.sync?

We could use a storage wrapper that falls back to local storage if storage.sync doesn't exist but I'm not a Brave user so it may take awhile to test.

**Comment by nanaya** [comment]
could this be the same reason why my settings are reset when restarting in chrome and edge as I'm not logged in to their respective sync system?

**Comment by melink14** [comment]
That's possible. I would have hoped that the sync storage API would work locally if syncing was turned off but we may not be lucky.

That should be easier to test though since I can just use a new profile with no account.

**Comment by nanaya** [comment]
Nevermind, my specific problem is because this line:

https://github.com/melink14/rikaikun/blob/bb011dc9e825754bef36e89ab5d2cc58416cd2c7/extension/background.ts#L94

When the stored value is `false`, it'll always fallback to default.

**Comment by melink14** [comment]
Ah, a classic programming error, thanks for finding that.

I assume you noticed it was only happening to a couple of your settings (those with non false default)?
I opened https://github.com/melink14/rikaikun/issues/346 to track it explicitly.

**Comment by nanaya** [comment]
> I assume you noticed it was only happening to a couple of your settings (those with non false default)?

Yes. Thanks. On newer typescript replacing `||` with `??` should be enough to fix it.

**Comment by melink14** [comment]
Great tip; typescript migration is still barely started so it's missing common idioms like that!
